library/src/main/java/org/apache/apex/malhar/lib/state/managed/MovingBoundaryTimeBucketAssigner.java

```diff
@@ -125,9 +125,11 @@ public long getTimeBucket(long time)
       long move = (diffInBuckets + 1) * bucketSpanMillis;
       start += move;
       end += move;
-      lowestPurgeableTimeBucket += diffInBuckets;
       // trigger purge when lower bound changes
-      triggerPurge = (diffInBuckets > 0);
+      triggerPurge = (diffInBuckets >= 0);
```

Comment:
@tweise this change is required as the entire computation considers the interval [start, end) rather than [start, end]. So for [start, end), we need to trigger even if diff >= 0